listenbrainz/webserver/rabbitmq_connection.py

```diff
@@ -26,11 +26,44 @@ def init_rabbitmq_connection(app):
         credentials=pika.PlainCredentials(app.config['RABBITMQ_USERNAME'], app.config['RABBITMQ_PASSWORD']),
     )
 
-    _rabbitmq = pika_pool.QueuedPool(
-        create=lambda: pika.BlockingConnection(connection_parameters),
-        max_size=100,
-        max_overflow=10,
-        timeout=10,
-        recycle=3600,
-        stale=45,
-    )
+    _rabbitmq = RabbitMQConnectionPool(app.logger, connection_parameters, app.config['MAXIMUM_RABBITMQ_CONNECTIONS'])
+    _rabbitmq.add()
+    app.logger.error('Connection to RabbitMQ established!')
+
+
+class RabbitMQConnectionPool:
+    def __init__(self, logger, connection_parameters, max_size):
+        self.log = logger
+        self.connection_parameters = connection_parameters
+        self.max_size = max_size
+        self.queue = queue.Queue(maxsize=max_size)
+
+    def add(self):
+        try:
+            self.queue.put_nowait(self.create())
+        except queue.Full:
+            self.log.error('Tried to add a new connection into a full queue...', exc_info=True)
+
+    def get(self):
+        while True:
+            try:
+                connection, channel = self.queue.get_nowait()
+                if connection.is_open:
+                    return connection, channel
+                else:
+                    return self.create()
```

Comment:
Instead of creating a new connection right away if a connection is closed, I think it makes more sense to try and fetch the next possibly open connection in the queue and see if it is open and if so use it. That should reduce the overall number of connections used, no?